index.src.html

```diff
@@ -229,21 +249,18 @@
         {{InvalidStateError}}.
     2. Increment {{AudioDecoder/decodeQueueSize}}.
     3. <a>Queue a control message</a> to decode the |chunk|.
-    4. <a>Run the control message processing loop</a>.
 
     Running a control message to decode the chunk means performing these steps:
-    1. Decrement {{AudioDecoder/decodeQueueSize}}
-    2. Let |codec implementation queue| be the result of starting a new
-        <a>parallel queue</a>.
-    3. Enqueue the following steps to |codec implementation queue|:
-        1. Attempt to use {{AudioDecoder/[[codec implementation]]}} to decode
-            the chunk.
-        2. If decoding results in an error, queue a task on the media element
-            task source to run the <a>Close AudioDecoder</a> algorithm with
-            {{EncodingError}}.
-        3. Otherwise, for each decoded audio data |output|, queue a task on the
-            media element task source to run the <a>Output AudioFrame</a>
-            algorithm with |output|.
+    1. Attempt to use {{AudioDecoder/[[codec implementation]]}} to decode the
+        chunk.
+    2. If decoding results in an error, queue a task on the [=control thread=]
+        event loop to run the <a>Close AudioDecoder</a> algorithm with
+        {{EncodingError}}.
+    3. Queue a task on the [=control thread=] event loop to decrement
+        {{AudioDecoder/decodeQueueSize}}
+    4. Otherwise, for each decoded audio data |output|, queue a task on the
+        [=control thread=] event loop to run the <a>Output AudioFrame</a>
+        algorithm with |output|.
```

Comment:
Yes, this is what I had in mind during our chat yesterday evening. We mentioned that we could have 2 and 3 reversed, but that in practice it didn't really matter, did you settle on the order for a particular reason? Again, it's probably not too important.